src/CommonErrors.cpp

```diff
@@ -0,0 +1,14 @@
+#include "CommonErrors.hpp"
+
+#include "Range.hpp"
+
+namespace cmake::language
+{
+  tl::unexpected<Error> EmptyRangeError(const Range& range)
```

Comment:
I would name it `CreateEmptyRangeError`, `EmptyRangeError` sounds like a type name.